### PR TITLE
Add provider tests to github workflows

### LINKED ISSUE #414: Setup integration tests for BlsProvider & BlsSigner
Currently, the BlsProvider and BlsSigner tests exist in a new directory called `./clientsTests` within the `./contracts` directory. They are not included in any automated test run and require an aggregator instance to test against.

**Tasks**:
1. If needed, find a better home for the tests
2. Write commands to spin up environment to test against
3. Include them as part of integrations tests that can be automated in our workflows

Discussion from a [previous PR](https://github.com/web3well/bls-wallet/pull/402#discussion_r1033710118)

As part of this work, we should also:
1. Move chai spies set up to a config/init file
2. Integrate tests added as part of #385

### PR DESCRIPTION
**Pull request by JohnGuilding**
## What is this PR doing?
This PR adds the newly created Bls Provider and Signer tests to the github workflows and tests them against an aggregator instance.

**Other changes:**
- Added aggregator logs artifact (See below workflow run and scroll to the bottom of the page)
- Move chai config to single location
- Tests were failing unexpectedly due to reuse of state such as private keys. Each test now uses a new private key

## How can these changes be manually tested?
1. Go to actions and observe the [following run](https://github.com/web3well/bls-wallet/actions/runs/3831809306)
2. Check all tests pass in the `yarn test-integration` step

## Does this PR resolve or contribute to any issues?
#414 

## Checklist
- [x] I have manually tested these changes
- [x] Post a link to the PR in the group chat

## Guidelines
- If your PR is not ready, mark it as a draft
- The `resolve conversation` button is for reviewers, not authors
  - (But add a 'done' comment or similar)

### CODE REVIEW

**Comment by JohnGuilding** [comment]
Commit history is quite messy here as there were a lot of small, iterative changes when testing the github workflows. Plan would be to squash these commits when merging.

**Comment by JohnGuilding** [comment]
Got a weird issue where the one of the signer tests fails during the github workflow, but not locally. The test in question is "should send ETH (empty call) successfully".

I've tried moving the test from test 1 to test 3, but that had no effect.

Locally: All tests pass
Workflow: One test fails - https://github.com/web3well/bls-wallet/actions/runs/3695015162/jobs/6256895670

**Comment by JohnGuilding** [comment]
Will squash these commits into one as had to test the changes multiple times against github workflows. We also realised several pieces of state (private keys) were being shared across tests, so we removed/fixed these cases.

File: .github/actions/local-contract-deploy/action.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      shell: bash
+      run: yarn hardhat node &
+
+    - working-directory: ./contracts
+      shell: bash
+      run: yarn hardhat fundDeployer --network gethDev
+
+    - working-directory: ./contracts
+      shell: bash
+      run: yarn hardhat run scripts/deploy_all.ts --network gethDev
```
Comment (blakecduncan) [inline .github/actions/local-contract-deploy/action.yml]: nit but I would add a new line to the file

File: .github/workflows/clients.yml [hunk 2]
```diff
@@ -29,4 +32,27 @@ jobs:
...
+    - working-directory: ./contracts
+      run: yarn test-integration
+
+    # - name: upload artifacts
+    - uses: actions/upload-artifact@v3
+      if: always()
+      with:
+        name: my-artifact
+        path: ./aggregator/aggregatorLogs.txt
+        retention-days: 5
```
Comment (blakecduncan) [inline .github/workflows/clients.yml]: same here with the new line

File: contracts/test-integration/BlsProvider.test.ts [hunk 3]
```diff
@@ -474,36 +472,24 @@ describe("BlsProvider", () => {
...
   });
 });
 
 describe("JsonRpcProvider", () => {
   beforeEach(async () => {
-    signers = await hardhatEthers.getSigners();
     rpcUrl = "http://localhost:8545";
     regularProvider = new ethers.providers.JsonRpcProvider(rpcUrl);
-    regularSigner = regularProvider.getSigner();
+    // First two hardhat accounts are used in aggregator .env, which causes a nonce too low error when using the default signer here.
```
Comment (blakecduncan) [inline contracts/test-integration/BlsProvider.test.ts:491]: 👍

**Review by blakecduncan — APPROVED** [review]
lgtm 💯

**Review by kautukkundan — APPROVED** [review]
LGTM

**Pull request merged**